src/qt/diagnosticsdialog.cpp

```diff
@@ -168,386 +169,315 @@ void DiagnosticsDialog::UpdateOverallDiagnosticResult(DiagnosticResult diagnosti
-{
-    return m_researcher_model->hasEligibleProjects();
-}
-
-bool DiagnosticsDialog::VerifyCPIDIsEligible()
-{
-    return m_researcher_model->hasActiveBeacon();
-}
+    QString tooltip;
 
-bool DiagnosticsDialog::VerifyWalletIsSynced()
-{
-    return !OutOfSyncByAge();
-}
+    if (overall_diagnostic_result == warning)
+    {
+        tooltip = tr("One or more tests have generated a warning status. Wallet operation may be degraded. Please see "
+                     "the individual test tooltips for details and recommended action(s).");
+    }
+    else if (overall_diagnostic_result == failed)
+    {
+        tooltip = tr("One or more tests have failed. Proper wallet operation may be significantly degraded or impossible. "
+                     "Please see the individual test tooltips for details and recommended action(s).");
+    }
+    else if (overall_diagnostic_result == passed)
+    {
+        tooltip = tr("All tests passed. Your wallet operation is normal.");
+    }
+    else
+    {
+        tooltip = QString();
+    }
 
-bool DiagnosticsDialog::VerifyCPIDHasRAC()
-{
-    return m_researcher_model->hasRAC();
+    SetResultLabel(ui->overallResultResultLabel, overall_diagnostic_status, overall_diagnostic_result,
+                   QString(), tooltip);
 }
 
-double DiagnosticsDialog::VerifyETTSReasonable()
+void DiagnosticsDialog::on_testButton_clicked()
 {
-    // We are going to compute the ETTS with ignore_staking_status set to true
-    // and also use a 960 block diff as the input, which smooths out short
-    // term fluctuations. The standard 1-1/e confidence (mean) is used.
-
-    double diff = GRC::GetAverageDifficulty(960);
-
-    double result = GRC::GetEstimatedTimetoStake(true, diff);
+    // Check to see if there is already a test run in progress, and if so return.
+    // We do not want overlapping test runs.
 
-    return result;
-}
+    if (GetNumberOfTestsPending())
+    {
+        LogPrintf("INFO: DiagnosticsDialog::on_testButton_clicked: Tests still in progress from a prior run: %u",
+                  GetNumberOfTestsPending());
 
-int DiagnosticsDialog::VerifyCountSeedNodes()
-{
-    LOCK(cs_vNodes);
+        return;
+    }
 
-    int count = 0;
+    ResetOverallDiagnosticResult();
+    DisplayOverallDiagnosticResult();
 
-    for (auto const& vnodes : vNodes)
-        if (!vnodes->fInbound)
-            count++;
+    // Tests that are common to both investor and researcher mode.
 
-    return count;
-}
+    // wallet synced
+    VerifyWalletIsSynced();
 
-int DiagnosticsDialog::VerifyCountConnections()
-{
-    LOCK(cs_vNodes);
+    // connection count
+    unsigned int connections = CheckConnectionCount();
 
-    return (int)vNodes.size();
-}
+    // outbound connection count
+    CheckOutboundConnectionCount();
 
-void DiagnosticsDialog::on_testButton_clicked()
-{
-    // Check to see if there is already a test run in progress, and if so return.
-    // We do not want overlapping test runs.
+    // clock
```

Comment:
Haha. These were originally because the original function names were not in accordance with the tests. Since I have straightened all of that out, I agree.